src/DDTrace/Configuration/EnvVariableRegistry.php

```diff
@@ -8,6 +8,11 @@
  */
 class EnvVariableRegistry implements Registry
 {
+    /**
+     * @var array
+     */
+    private $keyTranslator = [];
```

Comment:
I am actually quite surprised that this brings a noticeable performance improvement. The reason for saying this is that this registry is a very low level one that is only called once per key (at least in theory). In fact, the upstream classes cache the results. Not saying we should not so it (actually some values are not cached on purpose) but still I am surprised :)

Reply:
Indeed. I've seen on the Profiler it being shown quite high. TBF I haven't checked this after changes if it improved the performance significantly. I've since now removed it when I noticed the boolValue et. al does indeed cache it. https://github.com/DataDog/dd-trace-php/pull/398#discussion_r274365118